### DISCUSSION TITLE
Bug in shapely.buffer

### DISCUSSION THREAD

**Discussion by laszipt** [shapely/shapely#1774] [Q&A]
The _**buffer points**_ calculation does not consider the location of the point on the map, so it becomes more and more elliptic as is it gets closer to the poles.

## Expected behavior and actual behavior.

It is expected to be a perfect circle around a single point.

## Steps to reproduce the problem.

```
from shapely import Point, buffer

point_on_equator = Point(-78.4556256102772,-0.002087481232010191)
circle   = point_on_equator.buffer(0.1)
#display resulted polygon in your preferred environment

point_in_alaska  = Point(-149.9941411317871,61.17072288201612)
ellipse  = point_in_alaska.buffer(0.1)
#display resulted polygon in your preferred environment
```

**Comment by brendan-ward** [comment]
`buffer` works in the same coordinate system as your points, so in your example it is `0.1` degrees around.  The circular buffer may become distorted if you reproject it into a different coordinate system (e.g., for display on a map); this is entirely expected.  See [Tissot's indicatrix](https://en.wikipedia.org/wiki/Tissot%27s_indicatrix) for visual examples of this.

Shapely is a planar geometry library and treats geographic coordinates just like any other planar coordinates; it does not create buffers based on a spheroid.

**Comment by laszipt** [comment]
Dear Ward,

Now I understand the way shapely is working, thanks to that.

Could you give me some hint on how to make a buffer on a spheroid or
project it in a different way ?


Thanks,
Tamás


Brendan Ward ***@***.***> ezt írta (időpont: 2023. febr. 24.,
P, 23:26):

**Comment by mwtoews** [comment]
There are several techniques to create a geodesic point buffer. Most of this work is not with shapely, but with other companion packages like [pyproj](https://pyproj4.github.io/pyproj/stable/).

A common technique is to use a project coordinate system, such as a UTM zone. For example, [see this answer](https://gis.stackexchange.com/a/327046/). This requires the data to fit nicely in one UTM zone, and may have distortions for large distances.

Another method is to use an [azimuthal equidistant projection](https://en.wikipedia.org/wiki/Azimuthal_equidistant_projection). For example, [see this answer](https://gis.stackexchange.com/a/289923/). This does not require any further projection information (except for choice of spheroid), and does not have any distortion, since it is equidistant from the buffer point.
